lib/theory/src/Theory/Constraint/Solver/Simplify.hs

```diff
@@ -265,6 +267,24 @@ enforceEdgeUniqueness = do
         merge _    []            = error "exploitEdgeProps: impossible"
         merge proj (keep:remove) = map (Equal (proj keep) . proj) remove
 
+    readBeforeConsume :: Reduction ChangeIndicator
```

Comment:
This function could use some explaining: what it is doing, but also how it is doing it.